commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountUserRelLocalServiceImpl.java

```diff
@@ -143,26 +147,45 @@ public void addCommerceAccountUserRels(
 	}
 
 	@Override
-	public void addDefaultRole(long userId, ServiceContext serviceContext)
+	public void addDefaultRoles(long userId, ServiceContext serviceContext)
 		throws PortalException {
 
 		User user = userPersistence.findByPrimaryKey(userId);
 
-		Role role = rolePersistence.fetchByC_N(
-			user.getCompanyId(),
-			CommerceAccountConstants.ROLE_NAME_ACCOUNT_USER);
-
-		if (role == null) {
-			role = roleLocalService.addRole(
-				user.getUserId(), null, 0,
-				CommerceAccountConstants.ROLE_NAME_ACCOUNT_USER,
-				Collections.singletonMap(
-					serviceContext.getLocale(),
-					CommerceAccountConstants.ROLE_NAME_ACCOUNT_USER),
-				Collections.emptyMap(), RoleConstants.TYPE_SITE,
-				StringPool.BLANK, serviceContext);
+		CommerceAccountServiceConfiguration
+			commerceAccountServiceConfiguration =
+				_configurationProvider.getSystemConfiguration(
+					CommerceAccountServiceConfiguration.class);
+
+		String[] siteRoles = commerceAccountServiceConfiguration.siteRoles();
+
+		if ((siteRoles == null) && ArrayUtil.isEmpty(siteRoles)) {
+			return;
 		}
 
+		Set<Role> roles = new HashSet<>();
+
+		for (String siteRole : siteRoles) {
+			Role role = rolePersistence.fetchByC_N(
```

Comment:
Can't we use `roleLocalService.fetchRole` here? That'd be better

Reply:
@Commando18 looks like this line is still the same, and SF doesn't allow it. Can you make the change?